src/transformers/tokenization_utils.py

```diff
@@ -1820,48 +1821,50 @@ def _convert_token_to_id_with_added_voc(self, token):
             return self.unk_token_id
         return id
 
-    def _convert_id_to_token(self, index):
+    def _convert_id_to_token(self, index: int):
         return self._tokenizer.id_to_token(int(index))
 
-    def convert_tokens_to_string(self, tokens):
-        return self._tokenizer.decode(tokens)
+    def convert_tokens_to_string(self, tokens: List[int], skip_special_tokens: bool = False) -> str:
+        return self._tokenizer.decode(tokens, skip_special_tokens)
 
-    def add_tokens(self, new_tokens):
+    def add_tokens(self, new_tokens: List[Union[str, Tuple[str, bool]]]) -> int:
         if isinstance(new_tokens, str):
             new_tokens = [new_tokens]
         return self._tokenizer.add_tokens(new_tokens)
 
-    def add_special_tokens(self, special_tokens_dict):
+    def add_special_tokens(self, special_tokens_dict: dict) -> int:
         added = super().add_special_tokens(special_tokens_dict)
         self._update_special_tokens()
         return added
 
-    def build_inputs_with_special_tokens(self, token_ids_0, token_ids_1=None):
+    def build_inputs_with_special_tokens(
+        self, token_ids_0: List[int], token_ids_1: Optional[List[int]] = None
+    ) -> List[int]:
         if token_ids_1 is None:
             return token_ids_0
         else:
             return token_ids_0 + token_ids_1
 
-    def num_special_tokens_to_add(self, pair=False):
+    def num_special_tokens_to_add(self, pair: bool = False) -> int:
         return self.tokenizer.num_special_tokens_to_add(pair)
 
-    def tokenize(self, text, pair: Optional[str] = None, add_special_tokens: bool = True):
+    def tokenize(self, text: str, pair: Optional[str] = None, add_special_tokens: bool = True) -> List[str]:
         return self.tokenizer.encode(text, pair, add_special_tokens).tokens
 
     def batch_encode_plus(
         self,
-        batch_text_or_text_pairs=None,
-        add_special_tokens=True,
-        max_length=None,
-        stride=0,
-        truncation_strategy="longest_first",
-        pad_to_max_length=False,
-        return_tensors=None,
-        return_token_type_ids=True,
-        return_attention_mask=True,
-        return_overflowing_tokens=False,
-        return_special_tokens_mask=False,
-        return_offsets_mapping=False,
+        batch_text_or_text_pairs: Union[List[str], List[Tuple[str, str]]] = None,
```

Comment:
Let's allow batch pretokenized inputs and change this to `Union[List[str], List[List[str]],, List[Tuple[str, str]], List[Tuple[List[str], List[str]]]]`?

Reply:
I think we can assume that if the inputs are sentence pairs, both members of the pairs will be either not pretokenized or both pretokenized.

Reply:
How do we want to handle pairs here ? On Rust side, we have to provide the a list of first tuples part and then a second one: 

```python
# First part
enc_first = encode_tokenized_batch([s for (s, p) in batch_text_or_text_pairs], 0)

# Second part
enc_second = encode_tokenized_batch([p for (s, p) in batch_text_or_text_pairs], 1)
```

Then we have two BatchEncoding, unrelated to each other (@n1t0 correct me if I'm wrong here), do we want to unify them ? Return both ?